consensus/state_processing/src/per_block_processing/process_operations.rs

```diff
@@ -447,6 +515,280 @@ pub fn apply_deposit<E: EthSpec>(
         if let Ok(inactivity_scores) = state.inactivity_scores_mut() {
             inactivity_scores.push(0)?;
         }
+
+        // [New in Electra:EIP7251]
+        if let Ok(pending_balance_deposits) = state.pending_balance_deposits_mut() {
+            pending_balance_deposits.push(PendingBalanceDeposit {
+                index: new_validator_index as u64,
+                amount,
+            })?;
+        }
+    }
+
+    Ok(())
+}
+
+pub fn process_execution_layer_withdrawal_requests<E: EthSpec>(
+    state: &mut BeaconState<E>,
+    requests: &[ExecutionLayerWithdrawalRequest],
+    spec: &ChainSpec,
+) -> Result<(), BlockProcessingError> {
+    for request in requests {
+        let amount = request.amount;
+        let is_full_exit_request = amount == spec.full_exit_request_amount;
+
+        // If partial withdrawal queue is full, only full exits are processed
+        if state.pending_partial_withdrawals()?.len() == E::pending_partial_withdrawals_limit()
+            && !is_full_exit_request
+        {
+            continue;
+        }
+
+        // Verify pubkey exists
+        let index_opt = state.get_validator_index(&request.validator_pubkey)?;
+        let Some(index) = index_opt else {
+            continue;
+        };
+
+        let validator = state.get_validator(index)?;
+
+        // Verify withdrawal credentials
+        let has_correct_credential = validator.has_execution_withdrawal_credential(spec);
+        let is_correct_source_address = validator
+            .get_execution_withdrawal_address(spec)
+            .map(|addr| addr == request.source_address)
+            .unwrap_or(false);
+
+        if !(has_correct_credential && is_correct_source_address) {
+            continue;
+        }
+
+        // Verify the validator is active
+        if !validator.is_active_at(state.current_epoch()) {
+            continue;
+        }
+
+        // Verify exit has not been initiated
+        if validator.exit_epoch != spec.far_future_epoch {
+            continue;
+        }
+
+        // Verify the validator has been active long enough
+        if state.current_epoch()
+            < validator
+                .activation_epoch
+                .safe_add(spec.shard_committee_period)?
+        {
+            continue;
+        }
+
+        let pending_balance_to_withdraw = state.get_pending_balance_to_withdraw(index)?;
```

Comment:
`get_pending_balance_to_withdraw` iterates over the full list of withdrawals making it a O(n^2) operation technically. We process only a few withdrawal requests per block, but we should watch to ensure we don't need a cache.

Reply:
maybe we should add per-operations metrics in a separate pr?

Reply:
https://github.com/sigp/lighthouse/issues/5999